### PR TITLE
skip when passed no translation

### PR DESCRIPTION
**Pull request by mkangia**
https://manage.dimagi.com/default.asp?277877#1505929

Just like we ensure at least one translation for [case properties](https://github.com/dimagi/commcare-hq/blob/d0067647e5872b7bf59bb1d2bfc05917a45354eb/corehq/apps/app_manager/app_translations/app_translations.py#L876), 
just added a check to ensure at least one translation for form labels for registration forms only.

### CODE REVIEW

**Review by pr33thi — APPROVED** [review]

File: corehq/apps/app_manager/app_translations/app_translations.py [hunk 1]
```diff
@@ -637,14 +637,29 @@ def is_markdown_vetoed(text_node_):
...
             vetoes[label_id] = vetoes[label_id] or is_markdown_vetoed(text_node)
             markdowns[label_id] = markdowns[label_id] or had_markdown(text_node)
-
+    # skip labels that have no translation provided
+    skip_label = set()
+    for row in rows:
+        for lang in app.langs:
+            if row.get(_get_col_key('default', lang)):
+                break
+        else:  # https://stackoverflow.com/a/654002
```
Comment (orangejenny) [inline corehq/apps/app_manager/app_translations/app_translations.py]: Linking to SO to explain a construct is a code smell. Why not just rewrite using an inner function?

Comment (mkangia) [reply corehq/apps/app_manager/app_translations/app_translations.py]: Cool, updated.

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
test failures look real

**Comment by mkangia** [comment]
Thanks @orangejenny, fixed!

**Review by orangejenny — APPROVED** [review]
thanks for adding a new test

**Comment by mkangia** [comment]
the build is failing, so still need to look at this.. added the label till then.

**Review by pr33thi — APPROVED** [review]

File: corehq/apps/app_manager/app_translations/app_translations.py [hunk 2]
```diff
@@ -645,15 +645,16 @@ def has_translation(row_, langs):
...
-    for row in rows:
-        if not has_translation(row, app.langs):
-            skip_label.add(row['label'])
-    for label in skip_label:
-        msgs.append((
-            messages.error,
-            "You must provide at least one translation" +
-            " for the label '%s' in sheet '%s'" % (label, sheet.worksheet.title)
-        ))
+    if form.is_registration_form():
```
Comment (pr33thi) [inline corehq/apps/app_manager/app_translations/app_translations.py]: how come this is only for registration forms?

Comment (mkangia) [reply corehq/apps/app_manager/app_translations/app_translations.py]: @pr33thi 
the failure only happens for registration forms
https://github.com/dimagi/commcare-hq/blob/52897b9680906ff5b4ad0954faac779bcdc249ab/corehq/apps/app_manager/util.py#L163 when `form.get_questions([app.default_language])` is called.

Comment (pr33thi) [reply corehq/apps/app_manager/app_translations/app_translations.py]: cool, thanks

**Pull request merged**